### PR TITLE
- Created `User` and `Role` entities

### PR DESCRIPTION
**Pull request by Likh-Alex**
- Created `Repository` and `Service` for `User` and `Role`

### CODE REVIEW

File: src/main/java/com/dev/bankingservice/entity/Role.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Enumerated(EnumType.STRING)
+    @Column(name = "role_name")
+    private RoleName roleName;
+
+    public enum RoleName {
+        CUSTOMER,
+        MANAGER
```
Comment (vlykhvar) [inline src/main/java/com/dev/bankingservice/entity/Role.java]: RoleName should be ADMIN and USER by requirements

**Review by vlykhvar — APPROVED** [review]

File: src/main/java/com/dev/bankingservice/entity/Role.java (hunk 1 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/entity/Role.java]: Thank you, renamed the role to `ADMIN` and `USER`.

File: src/main/java/com/dev/bankingservice/repository/RoleRepository.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.dev.bankingservice.repository;
+
+import com.dev.bankingservice.entity.Role;
+import javax.persistence.Id;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface RoleRepository extends JpaRepository<Role, Id> {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/bankingservice/repository/RoleRepository.java]:
```suggestion
public interface RoleRepository extends JpaRepository<Role, Long> {
```

File: src/main/java/com/dev/bankingservice/repository/RoleRepository.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    @Query("SELECT r FROM Role r WHERE r.roleName = :name")
+    Role getByName(@Param("name") Role.RoleName name);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/bankingservice/repository/RoleRepository.java]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

File: src/main/java/com/dev/bankingservice/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import com.dev.bankingservice.entity.User;
+
+public interface UserService {
+    User create(User user);
+
+    User getById(Long id);
+
+    User getByPhoneNumber(String phoneNumber);
+
+    void update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/bankingservice/service/UserService.java]: see checklist

File: src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @Override
+    public void update(User user) {
+        userRepository.save(user);
+    }
+
+    @Override
+    public void delete(Long id) {
+        userRepository.delete(userRepository.getOne(id));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java]:
```suggestion
        userRepository.deleteById(id);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/bankingservice/repository/RoleRepository.java (hunk 2 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/repository/RoleRepository.java]: Done

File: src/main/java/com/dev/bankingservice/repository/RoleRepository.java (hunk 3 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/repository/RoleRepository.java]: Thank you, done

File: src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java (hunk 5 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java]: Done

File: src/main/java/com/dev/bankingservice/service/UserService.java (hunk 4 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/service/UserService.java]: removed `update` method and rename `create` to `save` to use for both `create` and `update` operations.

File: src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public User getByPhoneNumber(String phoneNumber) {
+        return userRepository.getByPhoneNumber(phoneNumber).orElseThrow(()
+                -> new RuntimeException("Can not get user by phone number: " + phoneNumber));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        userRepository.delete(userRepository.getOne(id));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java]:
```suggestion
        userRepository.deleteById(id);
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java (hunk 5 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java]: probably you forgot to push the change here?

Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java]: Certainly, fixed it now

File: src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java (hunk 6 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java]: Now it is correct

**Pull request merged**